### PR TITLE
Feature/add username suggestions

### PR DESCRIPTION
**Pull request by theck13**
Add a method for fetching username suggestions for a WordPress.com account, `fetchUsernameSuggestions`.  It requires the `name` parameter using the `/wpcom/v2/users/username/suggestions` unauthenticated endpoint.

### CODE REVIEW

**Comment by nbradbury** [comment]
@theck13 This looks good, but do you think we should add a test to fetch suggestions for a non-empty name? Right now the only test is fetching them for an empty name.

**Comment by theck13** [comment]
> do you think we should add a test to fetch suggestions for a non-empty name?

That's good idea.  I added a successful test in 4009e3ebdcbf33854d2ab2cd55f3f7f56c510415.

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/account/AccountRestClient.java [hunk 1]
```diff
@@ -218,6 +233,43 @@ public void onErrorResponse(@NonNull BaseNetworkError error) {
...
+     *
+     * @param name  Text (e.g. display name) from which to create username suggestions
+     */
+    public void fetchUsernameSuggestions(@NonNull String name) {
+        String url = WPCOMV2.users.username.suggestions.getUrl();
+
+        Map<String, String> params = new HashMap<>();
+        params.put("name", name);
+
+        add(WPComGsonRequest.buildGetRequest(url, params, UsernameSuggestionsResponse.class,
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/account/AccountRestClient.java]: Does this need to be an authed request @theck13? Or can we use `addUnauthedRequest`?

Comment (theck13) [reply fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/account/AccountRestClient.java]: It _can_ in fact be an unauthenticated request.  I updated the method in ec186b035a9719daddb2e036e7f91cf80c804df3.

**Comment by nbradbury** [comment]
:shipit:
